src/components/useAddToFavorites.js

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+import { useParams } from 'react-router-dom';
```

Comment:
It's best to use react-router to handle URL params:
https://reactrouter.com/web/example/url-params